core/tests/protractor_desktop/topicAndStoryViewer.js

```diff
@@ -196,7 +197,19 @@ describe('Topic and Story viewer functionality', function() {
       await topicViewerPage.get('math', 'Topic TASV1');
       await topicViewerPage.moveToPracticeTab();
       await topicViewerPage.selectSkillForPractice('Subtopic TASV1');
-      await topicViewerPage.startPractice();
+
+      await waitFor.clientSideRedirection(async() => {
+        // Start practice to trigger redirection to practice session page.
+        await topicViewerPage.startPractice();
+      }, (url) => {
+        // Wait until the URL has changed to /practice.
+        return (/practice/.test(url));
+      }, async() => {
+        var practiceSessionContainer = element(
+          by.css('.protractor-test-practice-session-container'));
+        await waitFor.presenceOf(practiceSessionContainer);
```

Comment:
We try to avoid having element selectors in the `protractor_desktop` files. They should instead be in the utility files under `protractor_utils/`. You could create a `waitForPracticeSessionContainer` function in the utility file that gets called here instead

Reply:
done